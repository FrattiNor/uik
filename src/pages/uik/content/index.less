@import '~@/theme';

.doc {
    flex-grow: 1;
    padding: 0 48px 32px 48px;
    background-color: #fff;
    color: rgba(0, 0, 0, 0.85);
    font-size: 14px;
    position: relative;
    z-index: 1;
    min-width: 0;

    // 针对特定 自定义样式
    :global {
        // 只针对api table
        .api {
            
            td:nth-child(1) {
                font-weight: bold;
                width: 15%;
            }

            td:nth-child(2) {
                width: 25%;
            }

            td:nth-child(3) {
                color: #c41d7f;
                width: 35%;
            }

            td:nth-child(4) {
                width: 15%;
            }

            td:nth-child(5) {
                width: 10%;
            }
        }
    }
}
