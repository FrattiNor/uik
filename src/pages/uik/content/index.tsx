import React, { FC } from 'react'
import { useLocation } from 'react-router-dom'
import FormatMarkDown from '@/components/format_markdown'
import { getDoc } from '@/utils/doc'
import styles from './index.less'

const Content: FC = () => {
    const location = useLocation()
    const { pathname } = location
    const selectedName = pathname.replace('/uik/', '')
    const MD = getDoc(selectedName)

    return (
        <FormatMarkDown className={styles['doc']}>
            {MD && <MD />}
        </FormatMarkDown>
    )
}

export default Content
