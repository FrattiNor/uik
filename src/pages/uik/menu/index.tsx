import React, { FC, Fragment, useEffect, useState } from 'react'
import classnames from 'classnames'
import { useHistory, useLocation } from 'react-router-dom'
import Title from 'react-document-title'
import Sticky from 'react-sticky-el'
import { menu, getTitle, judgeName } from '@/doc_menu'
import styles from './index.less'

const Menu: FC = () => {
    const history = useHistory()
    const location = useLocation()

    const { pathname } = location
    const selectedName = pathname.replace('/uik/', '')

    const [pageTitle, setTitle] = useState('')

    // 跳转函数
    const push = (name: string): void => {
        window.scrollTo(0, 0)
        history.push(`/uik/${name}`)
    }

    // 如果没有选择，默认选择第一个
    useEffect(() => {
        if (!judgeName(selectedName)) {
            const FirstName = menu[0]?.components[0]?.name
            if (FirstName) push(FirstName)
        }
    }, [selectedName])

    // 设置title
    useEffect(() => {
        const title = getTitle(selectedName)
        setTitle(title)
    }, [selectedName])

    return (
        <Title title={pageTitle}>
            <Sticky className={styles['menu']}>
                {menu.map(({ title, components }) => {
                    return (
                        <Fragment key={title}>
                            <div className={styles['title']}>{title}</div>
                            <>
                                {components.map(({ desc, name }) => (
                                    <div
                                        onClick={(): void => push(name)}
                                        key={name}
                                        className={classnames(styles['text'], { [styles['selected']]: selectedName === name })}
                                    >
                                        {desc}
                                    </div>
                                ))}
                            </>
                        </Fragment>
                    )
                })}
            </Sticky>
        </Title>
    )
}

export default Menu
