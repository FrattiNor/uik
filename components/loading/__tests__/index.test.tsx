import React from 'react'
import { render } from '@testing-library/react'
import { Loading } from 'uik'

describe('<Loading />', () => {
    test('should render default', () => {
        const { container } = render(
            <Loading>
                <div>loading</div>
            </Loading>
        )
        expect(container).toMatchSnapshot()
    })

    test('should render loading', () => {
        const { container } = render(
            <Loading loading>
                <div>loading</div>
            </Loading>
        )
        expect(container).toMatchSnapshot()
    })
})
