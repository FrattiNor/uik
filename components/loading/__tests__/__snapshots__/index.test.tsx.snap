// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render button with type 1`] = `
<button
  class="uik-btn uik-btn-default uik-btn-middle"
  type="button"
>
  default
</button>
`;

exports[`<Button /> should render button with type 2`] = `
<button
  class="uik-btn uik-btn-default uik-btn-middle"
  type="button"
>
  primary
</button>
`;

exports[`<Button /> should render button with type 3`] = `
<button
  class="uik-btn uik-btn-default uik-btn-middle"
  type="button"
>
  danger
</button>
`;

exports[`<Button /> should render default 1`] = `
<div>
  <button
    class="uik-btn uik-btn-default uik-btn-middle"
    type="button"
  >
    default
  </button>
</div>
`;

exports[`<Loading /> should render default 1`] = `
<div>
  <div>
    loading
  </div>
</div>
`;

exports[`<Loading /> should render loading 1`] = `
<div>
  <div
    class="uik-loading"
    style="display: block;"
  >
    <div>
      <div
        class="uik-loading-mask"
      >
        <i
          class="uik-iconfont uik-icon-loading uik-loading-icon"
          style="font-family: uik-iconfont;"
        />
      </div>
    </div>
    <div
      class="uik-loading-content"
    >
      <div>
        loading
      </div>
    </div>
  </div>
</div>
`;
